File: mlua-gen/tests/function_of_struct.lua
local human = Human.default()
assert(human[1] ~= nil)
assert(human[1] == "")
human[1] = "heart"
assert(human[1] == "heart")
local martin = Human.new_martin()
local martin2 = Human("Martin")
assert(martin[1] == martin2[1])

File: mlua-gen/tests/function_of_struct.rs
use mlua_gen::{LuaBuilder, mlua_gen};

#[test]
pub fn test() {
    #[derive(Default)]
    #[mlua_gen(impl = [default(), new_martin()])]
    struct Human(pub(crate) String);

    impl Human {
        fn new_martin() -> Self {
            Human(String::from("Martin"))
        }
    }

    let lua = mlua::Lua::new();
    Human::to_globals(&lua).unwrap();

    lua.load(include_str!("./function_of_struct.lua"))
        .exec()
        .unwrap();
}
